# Import libraries

In [66]:
import pandas as pd
import datetime as dt

# Import data

In [113]:
data = pd.read_csv(r'C:/Users/tom3r/OneDrive/Desktop/GCCR003_pre-pst-merge_deID_dd_dn_manualcln_v1.csv',encoding= 'unicode_escape')

# Pre-Process

In [58]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3392 entries, 0 to 3391
Columns: 189 entries, Unnamed: 0 to Unnamed: 0.1
dtypes: float64(99), int64(7), object(83)
memory usage: 4.9+ MB


In [10]:
data.describe()

Unnamed: 0  Sample_Set_Number_x  Year_of_birth       Gender  \
count  3392.000000          3392.000000    3392.000000  3392.000000   
mean   4369.319575          4178.463443    1976.589328     0.283019   
std    1124.888664          2489.952151      12.799326     0.467871   
min       0.000000             2.000000    1936.000000     0.000000   
25%    3446.750000          2053.500000    1967.000000     0.000000   
50%    4393.500000          4099.500000    1977.000000     0.000000   
75%    5318.250000          5912.000000    1987.000000     1.000000   
max    6235.000000          9967.000000    2001.000000     3.000000   

       Respiratory_Illness_y/n  COVID_diagnosis_x  \
count              3392.000000        3362.000000   
mean                  0.991156           2.841761   
std                   0.093641           1.748363   
min                   0.000000           1.000000   
25%                   1.000000           1.000000   
50%                   1.000000           2.000000   
75%                   1.000000           4.000000   
max                   1.000000           8.000000   

       Other_viral_or_respiratory_illnesses_strep_throat_(streptococcal_bacteria)  \
count                                        3362.000000                            
mean                                            0.012790                            
std                                             0.112384                            
min                                             0.000000                            
25%                                             0.000000                            
50%                                             0.000000                            
75%                                             0.000000                            
max                                             1.000000                            

       Other_viral_or_respiratory_illnesses_another_bacterial_illness  \
count                                        3362.000000                
mean                                            0.010708                
std                                             0.102939                
min                                             0.000000                
25%                                             0.000000                
50%                                             0.000000                
75%                                             0.000000                
max                                             1.000000                

       Other_viral_or_respiratory_illnesses_flu_(influenza)  \
count                                        3362.000000      
mean                                            0.020226      
std                                             0.140794      
min                                             0.000000      
25%                                             0.000000      
50%                                             0.000000      
75%                                             0.000000      
max                                             1.000000      

       Other_viral_or_respiratory_illnesses_another_viral_illness  ...  \
count                                        3362.000000           ...   
mean                                            0.014575           ...   
std                                             0.119860           ...   
min                                             0.000000           ...   
25%                                             0.000000           ...   
50%                                             0.000000           ...   
75%                                             0.000000           ...   
max                                             1.000000           ...   

       Symptoms_loss_of_appetite_y  Symptoms_headache_y  \
count                  3254.000000          3254.000000   
mean                      0.054702             0.138291   
std                       0.227432             0.34525

# Relevant column names:



Date_of_onset ,

Smell_before_illness,

Blocked_nose_before_illness,

Taste_before_illness,

Chemethesis_before_illness,

FOLLOWUP

Smell_current

Taste_current

Email_Time_Stamp_y


# Percentages of recovery

In [118]:
data['percentage_recovery_smell'] = data['Smell_current'] / data['Smell_before_illness'] * 100
data['percentage_recovery_taste'] = data['Taste_current'] / data['Taste_before_illness'] * 100
len(data)

3392

In [119]:
data_excluded_prior = data.copy()

In [ ]:
Excluding cases with prior 

In [120]:
data_excluded_prior = data.query('Smell_before_illness > 50')
data_excluded_prior = data_excluded_prior.query('Taste_before_illness > 50')

In [121]:
len(data_excluded_prior)

3212

In [16]:
data['percentage_recovery_smell'].head()

0     34.636872
1     30.171544
2     48.419980
3     99.600000
4    101.522843
Name: percentage_recovery_smell, dtype: float64

# Long haulers count
Number of individuals with changes (under 80%)

<u> smell </u>

In [132]:
len(data_excluded_prior.query('percentage_recovery_smell < 80'))

1709

In [133]:
len(data_excluded_prior.query('percentage_recovery_smell < 70'))

1360

In [134]:
len(data_excluded_prior.query('percentage_recovery_smell > 100'))


414

interestingly there are 414 cases with higher percentage than the beggining

Number of individuals with changes (under 80%)

<u> taste </u>

In [138]:
len(data_excluded_prior.query('percentage_recovery_taste < 80'))

1183

In [139]:
len(data_excluded_prior.query('percentage_recovery_taste < 70'))

870

In [140]:
len(data_excluded_prior.query('percentage_recovery_taste > 100'))

657

here there are 657 (!) cases

# DATE DELTA

In [141]:
data_excluded_prior['Date_of_onset'] = pd.to_datetime(data_excluded_prior['Date_of_onset'])
data_excluded_prior['Email_Time_Stamp_y'] = pd.to_datetime(data_excluded_prior['Email_Time_Stamp_y'])

In [142]:
data_excluded_prior['days_from_onset'] = ((data_excluded_prior['Email_Time_Stamp_y'] - data_excluded_prior['Date_of_onset']).dt.total_seconds()/(60*60*24))

In [143]:
data_excluded_prior['days_from_onset'].head()

0    194.604861
1    182.605556
2    169.606250
3    203.608333
4    175.609028
Name: days_from_onset, dtype: float64